### PR TITLE
Make sure to dispose HttpContentStream when done reading module logs.

### PR DESCRIPTION
**Pull request by vadim-kovalyov**
This another attempt to find and fix a potential issue where we don't dispose sockets in time.

For `GetModuleLogs` operation, we provide `HttpCompletionOption.ResponseHeadersRead` when calling `HttpClient.SendAsync()`. In this case, the response is not loaded into memory at once, but instead can be streamed. It implies, that we also need to properly dispose the stream so the connection is released ASAP.

### CODE REVIEW

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/logs/LogsProvider.cs [hunk 1]
```diff
@@ -31,11 +31,13 @@ public async Task<byte[]> GetLogs(string id, ModuleLogOptions logOptions, Cancel
                 logOptions.Filter.IncludeTimestamp = Option.Some(true);
             }
 
-            Stream logsStream = await this.runtimeInfoProvider.GetModuleLogs(id, false, logOptions.Filter.Tail, logOptions.Filter.Since, logOptions.Filter.Until, logOptions.Filter.IncludeTimestamp, cancellationToken);
-            Events.ReceivedStream(id);
+            using (Stream logsStream = await this.runtimeInfoProvider.GetModuleLogs(id, false, logOptions.Filter.Tail, logOptions.Filter.Since, logOptions.Filter.Until, logOptions.Filter.IncludeTimestamp, cancellationToken))
```
Comment (arsing) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/logs/LogsProvider.cs:34]: Is there any convention on whether the `using` should be in the code that creates the `IDisposable` (here) or the code that "consumes" the `IDisposable` (`GetProcessedLogs`) ?

Comment (vadim-kovalyov) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Core/logs/LogsProvider.cs:34]: If I got your question right, there is no convention. In our case we pass the "active" `Stream` up the stack, so we pass the responsibility of disposing the stream to the "consumer".

File: edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/ProxyReadinessProbe.cs [hunk 2]
```diff
@@ -25,8 +25,10 @@ async Task<ProxyReadiness> CheckAsync(CancellationToken token)
         {
             try
             {
-                HttpResponseMessage response = await this.client.GetAsync($"/systeminfo?api-version={this.apiVersion}", token);
-                return response.StatusCode == HttpStatusCode.OK ? ProxyReadiness.Ready : ProxyReadiness.Failed;
+                using (HttpResponseMessage response = await this.client.GetAsync($"/systeminfo?api-version={this.apiVersion}", token))
```
Comment (nyanzebra) [inline edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/ProxyReadinessProbe.cs:28]: I feel all of these should be able to be caught with some sort of code analytics thing... with some warning like "WRN: response resource not being handled..." Do we have something like that already? If so, are we not running it? If not, should we add something like that?

Comment (vadim-kovalyov) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/ProxyReadinessProbe.cs:28]: I don't have answer to that. Quick search showed this - https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca2000 - but this is not applicable to `System.IO.Stream`.

Comment (nyanzebra) [reply edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/ProxyReadinessProbe.cs:28]: Yeah not sure on what the tool might be, but I will look into this a bit more... as maybe we can further automate?

**Review by nyanzebra — APPROVED** [review]

**Comment by nyanzebra** [comment]
> 
> 
> This another attempt to find and fix a potential issue where we don't dispose sockets in time.
> 
> For `GetModuleLogs` operation, we provide `HttpCompletionOption.ResponseHeadersRead` when calling `HttpClient.SendAsync()`. In this case, the response is not loaded into memory at once, but instead can be streamed. It implies, that we also need to properly dispose the stream so the connection is released ASAP.

I am understanding this to mean we re doing trial and error?

**Review by varunpuranik — APPROVED** [review]

**Review by arsing — APPROVED** [review]

**Pull request merged**